## Diagnostic Part 3

## Part a

In [4]:
import pandas as pd
import numpy as np

In [3]:
# Graffity Dataset
graffity_df = pd.read_csv('311_Service_Requests_-_Graffiti_Removal.csv')

In [6]:
graffity_df.head(2)

Creation Date     Status Completion Date Service Request Number  \
0    09/02/2016  Completed      09/02/2016            16-06219390   
1    09/02/2016  Completed      09/02/2016            16-06219419   

  Type of Service Request What Type of Surface is the Graffiti on?  \
0        Graffiti Removal     Cement (Sidewalk, Alley, Wall, Curb)   
1        Graffiti Removal                         Wood - Unpainted   

  Where is the Graffiti located?    Street Address  ZIP Code  X Coordinate  \
0                          Front  917 N ELSTON AVE   60642.0  1.168480e+06   
1                          Fence  911 N ELSTON AVE   60642.0  1.168483e+06   

   Y Coordinate  Ward  Police District  Community Area  SSA   Latitude  \
0  1.906511e+06  27.0             12.0            24.0  NaN  41.899002   
1  1.906430e+06  27.0             12.0            24.0  NaN  41.898779   

   Longitude                                 Location  
0 -87.656474  (41.89900174269792, -87.65647358728224)  
1 -87.656468  (41.89877903468827, -87.65646849694056)

In [5]:
graffity_df.shape

(63386, 18)

What is the probability that a graffity call came from the the address 7500 S Wolcott Ave? Given that it's possible that the address isn't specified in this way in the data set, we will try different variations. 

In [7]:
graffity_df[graffity_df['Street Address'] == '7500 S Wolcott Ave']

Empty DataFrame
Columns: [Creation Date, Status, Completion Date, Service Request Number, Type of Service Request, What Type of Surface is the Graffiti on?, Where is the Graffiti located?, Street Address, ZIP Code, X Coordinate, Y Coordinate, Ward, Police District, Community Area, SSA, Latitude, Longitude, Location]
Index: []

In [10]:
graffity_df[graffity_df['Street Address'] == '7500 S WOLCOTT AVE']

Empty DataFrame
Columns: [Creation Date, Status, Completion Date, Service Request Number, Type of Service Request, What Type of Surface is the Graffiti on?, Where is the Graffiti located?, Street Address, ZIP Code, X Coordinate, Y Coordinate, Ward, Police District, Community Area, SSA, Latitude, Longitude, Location]
Index: []

Given that the address does not appear in the dataset, the probability that a graffiti call came from this address is 0. 

In [11]:
# Sanitation Dataset
sanitation_df = pd.read_csv('311_Service_Requests_-_Sanitation_Code_Complaints.csv')

In [12]:
sanitation_df.head(2)

Creation Date     Status Completion Date Service Request Number  \
0    01/02/2016  Completed      01/04/2016            16-00015398   
1    01/02/2016  Completed      01/04/2016            16-00018850   

     Type of Service Request What is the Nature of this Code Violation?  \
0  Sanitation Code Violation                                      Other   
1  Sanitation Code Violation                          Dog feces in yard   

           Street Address  ZIP Code  X Coordinate  Y Coordinate  Ward  \
0       6959 S PAXTON AVE   60649.0  1.192081e+06  1.859250e+06   5.0   
1  8243 S ST LAWRENCE AVE   60619.0  1.181671e+06  1.850471e+06   6.0   

   Police District  Community Area   Latitude  Longitude  \
0              3.0            43.0  41.768766 -87.571478   
1              6.0            44.0  41.744924 -87.609906   

                                   Location  
0  (41.768766169704875, -87.57147751207171)  
1    (41.74492409837845, -87.6099058679411)

In [13]:
sanitation_df.shape

(25147, 16)

In [14]:
sanitation_df[sanitation_df['Street Address'] == '7500 S Wolcott Ave']

Empty DataFrame
Columns: [Creation Date, Status, Completion Date, Service Request Number, Type of Service Request, What is the Nature of this Code Violation?, Street Address, ZIP Code, X Coordinate, Y Coordinate, Ward, Police District, Community Area, Latitude, Longitude, Location]
Index: []

In [15]:
sanitation_df[sanitation_df['Street Address'] == '7500 S WOLCOTT AVE']

Creation Date     Status Completion Date Service Request Number  \
4287    04/14/2016  Completed      04/22/2016            16-02502456   

        Type of Service Request What is the Nature of this Code Violation?  \
4287  Sanitation Code Violation                           Garbage in alley   

          Street Address  ZIP Code  X Coordinate  Y Coordinate  Ward  \
4287  7500 S WOLCOTT AVE   60620.0  1.164988e+06  1.854895e+06  17.0   

      Police District  Community Area   Latitude  Longitude  \
4287              6.0            71.0  41.757433 -87.670907   

                                     Location  
4287  (41.75743294325899, -87.67090742952253)

We have found one entry, therefore there was one request which came from 7500 S Wolcott Ave. Given that the dataset contains 25147 rows, the probability that a call came from this address is:

In [17]:
1/sanitation_df.shape[0]

3.9766174891637175e-05

In [18]:
# Potholes Dataset
potholes_df = pd.read_csv('311_Service_Requests_-_Pot_Holes_Reported.csv')

In [20]:
potholes_df.head(2)

CREATION DATE           STATUS COMPLETION DATE SERVICE REQUEST NUMBER  \
0    01/02/2016  Completed - Dup      01/04/2016            16-00013207   
1    01/02/2016  Completed - Dup      01/04/2016            16-00018079   

  TYPE OF SERVICE REQUEST CURRENT ACTIVITY MOST RECENT ACTION  \
0       Pothole in Street              NaN                NaN   
1       Pothole in Street              NaN                NaN   

   NUMBER OF POTHOLES FILLED ON BLOCK    STREET ADDRESS      ZIP  \
0                                 NaN   1700 E 130TH ST  60633.0   
1                                 NaN  6826 W GRAND AVE  60707.0   

   X COORDINATE  Y COORDINATE  Ward  Police District  Community Area  SSA  \
0  1.188692e+06  1.819603e+06  10.0              4.0            51.0  NaN   
1  1.130387e+06  1.915203e+06  36.0             25.0            18.0  NaN   

    LATITUDE  LONGITUDE                                 LOCATION  
0  41.660126 -87.585015  (41.66012576202311, -87.58501516002178)  
1  41.923699 -87.796327  (41.92369912420517, -87.79632671494232)

In [21]:
potholes_df.shape

(71953, 19)

In [22]:
potholes_df[potholes_df['STREET ADDRESS'] == '7500 S Wolcott Ave']

Empty DataFrame
Columns: [CREATION DATE, STATUS, COMPLETION DATE, SERVICE REQUEST NUMBER, TYPE OF SERVICE REQUEST, CURRENT ACTIVITY, MOST RECENT ACTION, NUMBER OF POTHOLES FILLED ON BLOCK, STREET ADDRESS, ZIP, X COORDINATE, Y COORDINATE, Ward, Police District, Community Area, SSA, LATITUDE, LONGITUDE, LOCATION]
Index: []

In [23]:
potholes_df[potholes_df['STREET ADDRESS'] == '7500 S WOLCOTT AVE']

CREATION DATE     STATUS COMPLETION DATE SERVICE REQUEST NUMBER  \
23363    05/02/2016  Completed      07/12/2016            16-03005208   

      TYPE OF SERVICE REQUEST CURRENT ACTIVITY MOST RECENT ACTION  \
23363       Pothole in Street    Final Outcome    Pothole Patched   

       NUMBER OF POTHOLES FILLED ON BLOCK      STREET ADDRESS      ZIP  \
23363                                 3.0  7500 S WOLCOTT AVE  60620.0   

       X COORDINATE  Y COORDINATE  Ward  Police District  Community Area  SSA  \
23363  1.164988e+06  1.854895e+06  17.0              6.0            71.0  NaN   

        LATITUDE  LONGITUDE                              LOCATION  
23363  41.757345 -87.671052  (41.75734524032, -87.67105156625242)

By the same logic as above, given that there is only one entry, and assuming equal probabilities, we find that:

In [24]:
1/potholes_df.shape[0]

1.3897961169096494e-05

In [25]:
# Vacant building datasets
vacant_df = pd.read_csv('311_Service_Requests_-_Vacant_and_Abandoned_Buildings_Reported.csv')

In [26]:
vacant_df.head(2)

SERVICE REQUEST TYPE SERVICE REQUEST NUMBER  \
0  Vacant/Abandoned Building            16-00017936   
1  Vacant/Abandoned Building            16-00027497   

  DATE SERVICE REQUEST WAS RECEIVED  \
0                        01/02/2016   
1                        01/03/2016   

  LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD).  \
0                                              Front                      
1                                              Front                      

   IS THE BUILDING DANGEROUS OR HAZARDOUS? IS BUILDING OPEN OR BOARDED?  \
0                                      NaN                         Open   
1                                      NaN                         Open   

  IF THE BUILDING IS OPEN, WHERE IS THE ENTRY POINT?  \
0                                FRONT-SIDE AND REAR   
1                                                NaN   

  IS THE BUILDING CURRENTLY VACANT OR OCCUPIED?  \
0                                        Vacant   
1                                        Vacant   

  IS THE BUILDING VACANT DUE TO FIRE?  \
0                               False   
1                               False   

  ANY PEOPLE USING PROPERTY? (HOMELESS, CHILDEN, GANGS)  \
0                                               True      
1                                              False      

                    ...                     ADDRESS STREET SUFFIX ZIP CODE  \
0                   ...                                        ST  60621.0   
1                   ...                                        ST  60629.0   

   X COORDINATE  Y COORDINATE  Ward  Police District  Community Area  \
0  1.170327e+06  1.865488e+06  16.0              7.0            68.0   
1  1.153522e+06  1.862608e+06  23.0              8.0            66.0   

    LATITUDE  LONGITUDE                                 Location  
0  41.786386 -87.651033  (41.78638640364335, -87.65103294840148)  
1  41.778832 -87.712726     (41.778832091928, -87.7127256067861)  

[2 rows x 23 columns]

In [27]:
vacant_df.shape

(5564, 23)

In [28]:
vacant_df.columns

Index(['SERVICE REQUEST TYPE', 'SERVICE REQUEST NUMBER',
       'DATE SERVICE REQUEST WAS RECEIVED',
       'LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD).',
       'IS THE BUILDING DANGEROUS OR HAZARDOUS?',
       'IS BUILDING OPEN OR BOARDED?',
       'IF THE BUILDING IS OPEN, WHERE IS THE ENTRY POINT?',
       'IS THE BUILDING CURRENTLY VACANT OR OCCUPIED?',
       'IS THE BUILDING VACANT DUE TO FIRE?',
       'ANY PEOPLE USING PROPERTY? (HOMELESS, CHILDEN, GANGS)',
       'ADDRESS STREET NUMBER', 'ADDRESS STREET DIRECTION',
       'ADDRESS STREET NAME', 'ADDRESS STREET SUFFIX', 'ZIP CODE',
       'X COORDINATE', 'Y COORDINATE', 'Ward', 'Police District',
       'Community Area', 'LATITUDE', 'LONGITUDE', 'Location'],
      dtype='object')

Given that there's no columns for street address, we'll isolate it by number first. 

In [30]:
vacant_df[vacant_df['ADDRESS STREET NUMBER'] == 7500]

SERVICE REQUEST TYPE SERVICE REQUEST NUMBER  \
453  Vacant/Abandoned Building            16-01222606   

    DATE SERVICE REQUEST WAS RECEIVED  \
453                        02/29/2016   

    LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD).  \
453                                                NaN                      

     IS THE BUILDING DANGEROUS OR HAZARDOUS? IS BUILDING OPEN OR BOARDED?  \
453                                      NaN                          NaN   

    IF THE BUILDING IS OPEN, WHERE IS THE ENTRY POINT?  \
453                                                NaN   

    IS THE BUILDING CURRENTLY VACANT OR OCCUPIED?  \
453                                           NaN   

    IS THE BUILDING VACANT DUE TO FIRE?  \
453                                 NaN   

    ANY PEOPLE USING PROPERTY? (HOMELESS, CHILDEN, GANGS)  \
453                                                NaN      

                      ...                     ADDRESS STREET SUFFIX ZIP CODE  \
453                   ...                                        DR  60619.0   

     X COORDINATE  Y COORDINATE  Ward  Police District  Community Area  \
453  1.180207e+06  1.855373e+06   6.0              6.0            69.0   

      LATITUDE  LONGITUDE                                 Location  
453  41.758408  -87.61512  (41.75840759130991, -87.61512035710996)  

[1 rows x 23 columns]

In [31]:
vacant_df.loc[453]

SERVICE REQUEST TYPE                                                                   Vacant/Abandoned Building
SERVICE REQUEST NUMBER                                                                               16-01222606
DATE SERVICE REQUEST WAS RECEIVED                                                                     02/29/2016
LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD).                                        NaN
IS THE BUILDING DANGEROUS OR HAZARDOUS?                                                                      NaN
IS BUILDING OPEN OR BOARDED?                                                                                 NaN
IF THE BUILDING IS OPEN, WHERE IS THE ENTRY POINT?                                                           NaN
IS THE BUILDING CURRENTLY VACANT OR OCCUPIED?                                                                NaN
IS THE BUILDING VACANT DUE TO FIRE?                                                             

This does not correspond to our address, so we can conclude the address does not appear in this dataset. Therefore, the probability that a phone call about vacant buildings came from this address is 0. 

## Part b

Given that there's only two neighbourhoods, a quick google search gives us the required ZIP Codes. 
Uptown zip codes are 60613, 60640.
Lawndale zip codes are 60608, 60623, 60624.

In [32]:
# Uptown
graffity_df[graffity_df["ZIP Code"] == 60613.0].shape

(1517, 18)

In [34]:
graffity_df[graffity_df["ZIP Code"] == 60640.0].shape

(1450, 18)

In [35]:
graffity_df.shape

(63386, 18)

There's a 4.7% chance that the call came from Uptown

In [36]:
((1517 + 1450)/63386) * 100

4.680844350487489

In [37]:
# Lawndale 
graffity_df[graffity_df["ZIP Code"] == 60608.0].shape

(4881, 18)

In [38]:
graffity_df[graffity_df["ZIP Code"] == 60623.0].shape

(2896, 18)

In [39]:
graffity_df[graffity_df["ZIP Code"] == 60624.0].shape

(238, 18)

There's 12.6% chance that the call came from Lawndale

In [40]:
((4881 + 2896 + 238)/63386) * 100

12.64474805162023

## Part c

Now assume that you don’t have access to all the raw data and you know the following things:
There are a total of 1000 calls, 600 from Englewood and 400 from Uptown. Of the 600 calls from Englewood, 100 of them are about Graffiti Removal. Of the 400 calls from Uptown, 160 are about Graffiti Removal. If a call comes about Graffiti Removal, how much more/less likely is it that the call came from Englewood versus Uptown?


This is a conditional probability problem. Basically, we must compare P(Englewood|Graffiti) and P(Uptown|Graffiti). 

In [43]:
# We have: 
englewood_probability = 600/1000
uptown_probability = 400/1000

englewood_graff = 100/600
englewood_other = 500/600

uptown_graff = 160/400
uptown_other = 240/400

In [44]:
# total probability that it was a graffiti is
graff_prob = (englewood_probability * englewood_graff) + (uptown_probability * uptown_graff)

In [45]:
# probability that it was Englewood given that it was a graffiti call
(englewood_probability * englewood_graff) / graff_prob

0.3846153846153846

In [46]:
# probability that it was Uptown given that it was a graffiti call 
(uptown_probability * uptown_graff) / graff_prob

0.6153846153846155

It is more likely that the call came from Uptown